src/Pecee/Http/Input/InputHandler.php

```diff
@@ -215,52 +245,78 @@ protected function parseInputItem(array $array): array
      * Find input object
      *
      * @param string $index
-     * @param array ...$methods
-     * @return IInputItem|array|null
+     * @param string ...$methods
+     * @return mixed
      */
     public function find(string $index, ...$methods)
     {
+        return $this->toValue($this->findItem($index, ...$methods));
+    }
+
+    /**
+     * @param string $index
+     * @param string ...$methods
+     * @return InputItem|null|array
+     */
+    public function findItem(string $index, ...$methods){
         $element = null;
 
         if (\count($methods) === 0 || \in_array(Request::REQUEST_TYPE_GET, $methods, true) === true) {
-            $element = $this->get($index);
+            $element = $this->get($index, null, true);
         }
 
         if (($element === null && \count($methods) === 0) || (\count($methods) !== 0 && \in_array(Request::REQUEST_TYPE_POST, $methods, true) === true)) {
-            $element = $this->post($index);
+            $element = $this->post($index, null, true);
+        }
+
+        if (($element === null && count($methods) === 0) || (count($methods) !== 0 && in_array('body', $methods, true))) {
+            $element = $this->body($index, null, true);
         }
 
         if (($element === null && \count($methods) === 0) || (\count($methods) !== 0 && \in_array('file', $methods, true) === true)) {
-            $element = $this->file($index);
+            $element = $this->file($index, null, true);
         }
 
         return $element;
     }
 
+    /**
+     * @param InputItem|array|null $value
+     * @return mixed
+     */
+    private function toValue($value)
+    {
+        if($value === null)
```

Comment:
You are right. This was mainly created for ajax submits where I set the param to a specific type in js. But you are right 👍🏻.
Yes we kan create single functions, but what do you think about just providing a function like
```
public function parseValue(){
        if($this->value === 'true')
            return true;
        else if($this->value === 'false')
            return false;
        else if($this->value === '')
            return null;
        else
            return $this->value;
}
```
or do you preferr adding both (single casting and just parsing)?

Reply:
The developer knows what object to expect, so I don't really think there's a point in parsing the value. Plus the parse could be wrong, floats can easily be mistaken for integers etc.

If we create a method for the most common types, we only need to cast the value in most cases:

```php
function getValueAsInteger(): int {
   return (int)$this->value;
}
```

Reply:
Okay, sure 👍🏻. But I will add a small validation. Yes the developers know what they want to recieve, but not every developer is validating the incoming data and this could lead to erros and security issues.

Reply:
I agree, plus warnings could be disabled or not clearly visible.

Maybe throw some kind of `InvalidCastException` etc. That way the developer will know the risks of casing incorrect values when calling the method and have to take that into account in his project. This is very similar to other languages.

```php
if(\is_int($this->value) === false) {
   throw new InvalidCastException('Failed to cast value. The value is not a valid integer.');
}
return (int)$this->value;
```

Reply:
👍🏻 Great idea